nilearn/plotting/surf_plotting.py

```diff
@@ -342,9 +342,9 @@ def _get_ticks_matplotlib(vmin, vmax, cbar_tick_format):
     n_ticks = 5
     # ...unless we are dealing with integers with a small range
     # in this case, we reduce the number of ticks
-    if cbar_tick_format == "%i" and vmax - vmin < n_ticks:
+    if cbar_tick_format == "%i" and vmax - vmin < n_ticks - 1:
         ticks = np.arange(vmin, vmax + 1)
-        n_ticks = len(ticks)
+        assert len(ticks) <= n_ticks
```

Comment:
I think `ticks` should be set to `np.arange(int(vmin), int(vmax) + 1)` since they should be integers, right?

Reply:
let's add a test rather than an assertion in the function itself. we don't want to crash the program just because there are too many ticks in a colorbar

Reply:
Should they be integers? I really don't know. That's not the case in the "large range" case either.

In any case, the proper range would be `[floor(vmin), ceil(vmax)]`.

Reply:
To me they should be integers because the if condition verifies that `cbar_tick_format=='%i'`.

Reply:
Agreed, but why is this not the case for larger ranges with `np.linspace(vmin, vmax, n_ticks)`?

Reply:
Forget my initial comment, I was wrong, we don't need to cast anything here.
Formatting happens there:

https://github.com/nilearn/nilearn/blob/6aaf8f94433dfaf2967d69c64c0f688d7e1c812d/nilearn/plotting/surf_plotting.py#L512-L515

Reply:
I added a new commit to enforce integer ticks if the format is `"%i"`.

Reply:
Ah, just read the previous comment. So the last commit is useless and can be removed.